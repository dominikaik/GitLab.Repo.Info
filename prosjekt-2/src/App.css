.App {
  text-align: center;
}

#headline {
  padding-left: 2vw;
  padding-right: 2vw;
}

@media only screen and (max-width: 600px) {
  #headline {
    padding-top: 1vh;
    padding-bottom: 1vh;
  }
}

.gridViewItem {
  border-radius: 10px;
}
