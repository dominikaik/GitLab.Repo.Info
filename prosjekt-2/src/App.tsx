import { ThemeContext } from "./contexts/ThemeContext"
import { useContext, useState } from "react";
import "./App.css";
import UserInput from "./components/UserInput";
import DarkMode from "./components/DarkMode";
import { GlobalStyles } from "@mui/material";

function App() {
  const {themes} = useContext(ThemeContext); 
  const [isDarkMode] = useState(false); 
  const theme = isDarkMode ? themes.dark : themes.light;

//   const toggleDarkMode = () => {
//     localStorage.setItem("isDarkMode", JSON.stringify(!isDarkMode));
//     setIsDarkMode(!isDarkMode); 
// };  

  return (
    <div className="App"
    // onChange={toggleDarkMode}
    style={{color: theme.textcolor, backgroundColor: theme.backgroundColor }}>
      <GlobalStyles styles={{ body: { backgroundColor: theme.backgroundColor } }} />
      <DarkMode />
      <UserInput />
    </div>
  );
}

export default App;
