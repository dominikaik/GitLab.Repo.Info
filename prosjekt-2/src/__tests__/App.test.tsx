import { render } from '@testing-library/react';
import App from '../App';

// it("renders page correctly", () => {
//   const {container} = render(<App/>); 
//   expect(container).toMatchSnapshot(); 
// }); 
