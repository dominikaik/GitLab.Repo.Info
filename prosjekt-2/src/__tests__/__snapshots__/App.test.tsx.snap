// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders page correctly 1`] = `
<div>
  <div
    class="App"
    style="color: black; background-color: rgb(231, 224, 233);"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium darkButton css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      style="color: grey;"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="Brightness4Icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div>
      <h3
        id="headline"
      >
        Nice to meet you! 
        <br />
         Please add a project id and an access token so I can show you some information. 
        ðŸ¥€
      </h3>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1gpk5at-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="send-link-field"
            id="send-link-field-label"
          >
            GitLab repo project id
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-g9m4e0-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-1u9li0h-MuiInputBase-input-MuiFilledInput-input"
              id="send-link-field"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-sizeSmall css-jv61v3-MuiInputAdornment-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                style="color: black;"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="HelpOutlineIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <br />
        <div
          class="MuiFormControl-root MuiTextField-root css-1gpk5at-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="access-token-field"
            id="access-token-field-label"
          >
            Access token
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
              id="access-token-field"
              type="text"
              value=""
            />
          </div>
        </div>
        <br />
        <button
          class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1irj245-MuiButtonBase-root-MuiButton-root"
          disabled=""
          id="access-token"
          style="background-color: rgb(141, 107, 148);"
          tabindex="-1"
          type="button"
        >
          Submit
        </button>
      </div>
    </div>
  </div>
</div>
`;
